server_admin/topics/clients/oidc/con-audience.adoc

```diff
@@ -0,0 +1,123 @@
+the access token and the ID token, since the access token will not contain the client for which the token is issued as an audience.
+
+If you need the client itself as an audience, see the
+<<_audience_hardcoded, hardcoded audience>> option. However, using the same client as both frontend and REST service is not recommended.
+====
+
+[[_audience_hardcoded]]
+===== Hardcoded audience
+
+When your service relies on realm roles or does not rely on the roles in the token at all, it can be useful to use a hardcoded audience. A hardcoded audience is a protocol mapper, that will add the client ID of the specified service client as an audience to the token.
+You can use any custom value, for example a URL, if you want to use a different audience than the client ID.
+
+You can add the protocol mapper directly to the frontend client. If the protocol mapper is added directly, the audience will be always added as well.
+
+For more control over the protocol mapper, you can create the protocol mapper on the dedicated client scope, which will be called for example `good-service`.
+
+.Audience Protocol Mapper
+image:{project_images}/audience_mapper.png[]
+
+* From the <<_client_installation, Installation tab>> of the `good-service` client, you can generate the adapter configuration and you can confirm that _verify-token-audience_ option will be set to `true`. This forces the adapter to verify the audience if you use this configuration.
+
+* You need to ensure that the `my-app` frontend client is able to request `good-service` as an audience in its tokens.
++
+On the `my-app` client:
++
+. Click the _Client Scopes_ tab.
+. Assign `good-service` as an optional (or default) client scope. 
++
+See <<_client_scopes_linking, Client Scopes Linking section>> for more details.
+
+* You can optionally <<_client_scopes_evaluate, Evaluate Client Scopes>> and generate an example access token. `good-service` will be added to the audience of the generated access token if `good-service` is included in the _scope_ parameter, when you assigned it as an optional client scope.
+
+* In your `my-app` application, you must ensure that the _scope_ parameter is used. The value `good-service` must be included when you want to issue the token for accessing `good-service`. 
++
+See:
++
+** link:{adapterguide_link}#_params_forwarding[parameters forwarding section] if your application uses the servlet adapter.
+** link:{adapterguide_link}#_javascript_adapter[javascript adapter section] if your application uses the javascript adapter.
+
+NOTE: Both the _Audience_ and _Audience Resolve_ protocol mappers add the audiences to the access token only, by default. The ID Token
+typically contains only a single audience, the client ID for which the token was issued. This is a requirement of the OpenID Connect specification. However, the access token does not necessarily have the client ID, which was the token issued for, unless the audience mappers added it.
```

Comment:
remove "only"
Change "This" to "Including only a single audience.."
Change ", which was the token issued for,"  to "for which the token was issued"